treeherder/etl/buildapi.py

```diff
@@ -62,32 +62,31 @@ def find_job_guid(self, build):
                     prop["branch"], build))
                 raise e
 
-        request_ids_str = ",".join(map(str, request_ids))
-        request_time_list = []
-
-        if type(request_times) == dict:
-            for request_id in request_ids:
-                request_time_list.append(
-                    request_times[str(request_id)])
-            request_times_str = ','.join(
-                map(str, request_time_list))
-        else:
-            request_times_str = str(request_times)
+        request_id = request_ids[-1]
+        request_time = request_times[str(request_id)]
 
         job_guid_data = {'job_guid': '', 'coalesced': []}
 
         if len(request_ids) > 1:
-            # coallesced job detected, generate the coalesced
-            # job guids
-            for index, r_id in enumerate(request_ids):
-                #skip if buildbot doesn't have a matching number of ids and times
-                if len(request_time_list) > index:
+            # coallesced job detected, generate the coalesced job guids
+
+            # delete the last request_id, which is the current job, because
+            # it was not, itself, coalesced.  The other requests were
+            # coalesced to it.
+            request_ids.pop()
+
+            # build the list of job_guids that were coalesced
+            for r_id in request_ids:
+                r_id = str(r_id)
+                try:
                     job_guid_data['coalesced'].append(
-                        common.generate_job_guid(
-                            str(r_id), request_time_list[index]))
+                        common.generate_job_guid(r_id, request_times[r_id]))
+                except KeyError:
```

Comment:
Yeah, they MAY always match.  But there was a check for it on the removed line 84 above, so I figured I'd better account for it.  Buildbot is a mysterious beast.  I was thinking someone must have hit that at some point.

I used the exception because it's considered more "Pythonic".  Though this is an admittedly trivial example, and I do sometimes use LBYL instead of EAFP, in this case, I did EAFP.  In Java, we usually would use LBYL, but I'm changing my ways...  :)

Here's an article by someone who feels strongly about it:
http://www.jeffknupp.com/blog/2013/02/06/write-cleaner-python-use-exceptions/

Though, if that seems harder to follow, I'll change it.